backstory/src/main/java/com/google/sps/servlets/ImageAnalysisServlet.java

```diff
@@ -0,0 +1,229 @@
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.cloud.vision.v1.Feature;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.cloud.vision.v1.ImageSource;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.protobuf.ByteString;
+import com.google.sps.servletData.AnalyzedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet for managing image analysis with Vision API and blobstore for uploads */
+@WebServlet("/image-analysis")
+public class ImageAnalysisServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Query to find all analyzed image entities.
+    Query query = new Query("analyzed-image");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<AnalyzedImage> analyzedImages = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String imageUrl = (String) entity.getProperty("imageUrl");
+      String labelsJsonArray = (String) ((Text) entity.getProperty("labelsJsonArray")).getValue();
+
+      analyzedImages.add(new AnalyzedImage(imageUrl, labelsJsonArray));
+    }
+
+    response.setContentType("application/json;");
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
+    String analyzedImagesJsonArray = gson.toJson(analyzedImages);
+    response.getWriter().println(analyzedImagesJsonArray);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Expecting a post request from Blobstore containing the data fields from the image-upload
+    // form. After having gone through Blobstore, the request will include the images uploaded. The
+    // form in the HTML will connect to the Blobstore URL, which encodes the images and then
+    // redirects the request to this Url. In this doPost, the images are analyzed with Vision API,
+    // and the analysis along with the image URL will be sent to datastore for permanent storage.
```

Comment:
I am actually not sure - since technically the doPost Javadoc is overwritten. This message is mostly giving some context into the data pipeline in order to make what is happening in the doPost more clear - it is not entirely about the doPost method itself (only the last line). Do you think it would be better if I did not overwrite the doPost documentation and instead rewrote it to include this information somehow?

Reply:
Looks like you can use `{@inheritDoc}` and then add to the inherited docs:
https://dzone.com/articles/inheriting-javadoc-method-comments

Reply:
Bump.

Reply:
Sounds good, I updated the documentation.